includes/class-woocommerce.php

```diff
@@ -227,6 +227,34 @@ private function define( $name, $value ) {
 		}
 	}
 
+	/**
+	 * Returns true if the request is a REST API request.
+	 *
+	 * @return bool
+	 */
+	private static function is_rest_api_request() {
+		$request_uri = $_SERVER['REQUEST_URI']; // @codingStandardsIgnoreLine
+		if ( empty( $request_uri ) ) {
+			return false;
+		}
+
+		// Legacy API.
+		if ( 0 === strpos( $request_uri, '/wc-api/' ) ) {
```

Comment:
This should work only when setting nice permalinks, probably better check only for `wc-api`.

Still IPN for payment gateways, like PayPal, will hit this endpoint too, so this may cause issues in payment gateways.

Reply:
It's easy check for `wc-apì` without need to check the entire URL.

```php
global $wp;

if ( ! empty( $wp->query_vars['wc-api'] ) || ! empty( $_GET['wc-api'] ) ) {
    return true;
}
```

>  can you maybe expand on in what way you think this might be a problem?

That's the problem, people can do anything inside that endpoint, you can build any URL for it, like `example.com/wp-api/hello-world`, and then just need to hook some function into `woocommerce_api_hello_world`, so there is many possibilities of changing it cause some issue, of course, it also may improve IPN response as well. I'm not saying to not do it, just pointing that this may raise backwards compatibility issues.

Reply:
@claudiosanches @peterfabian If this is a change for performance reasons, why don't we just target the NEW rest API here? We can leave cart present for legacy API and those speciality `wc-api` endpoints which gateways use.

Reply:
For me fine leave our legacy REST API outside, about backwards compatibility, probably a warning or blog post should help. Still this code is a copy from `WC_REST_Authentication->is_request_to_rest_api` so we probably could refactor this code, also there is a filter that has been copied, before it should be used for allow our authentication methods for more endpoints, but now will allow requests or not... So this need to be improved.

Reply:
If someone needs the cart in a plugin (like the one you found) we could also wrap the function with a filter too.

Reply:
A filter should be fine for plugins enable it.

Reply:
Yup, it would be nice to factor out the common code out, but (as we agreed) now that this new function will only be checking new REST API request, maybe it's not worth doing it?
The problem I see with the refactoring and extracting the code out is that it needs to be available in classes `WooCommerce` and `WC_REST_Authentication` and if we want to make it private in class `WooCommerce`, it's difficult to make it available in `WC_REST_Authentication`. 

Creating a separate class for this function to be available in both places seems like an overkill to me? 

So perhaps a private function in class `WooCommerce` with different filter name?

Reply:
I updated the code, so maybe you can have another look, @claudiosanches ?

Reply:
@peterfabian 

> now that this new function will only be checking new REST API request, maybe it's not worth doing it?

`WC_REST_Authentication` only works for our new REST API integration, not for our legacy REST API.

Reply:
> WC_REST_Authentication only works for our new REST API integration, not for our legacy REST API.
You're right, I missed that. Nevertheless, if we want to use the strategy 'private function for now, then swap for WP standard function later', it's not really possible to share the code, right?